pynestml/codegeneration/nest_reference_converter.py

```diff
@@ -228,6 +236,65 @@ def convert_name_reference(self, variable, prefix=''):
             NestNamesConverter.name(symbol) + \
             ('[i]' if symbol.has_vector_parameter() else '')
 
+    def __get_unit_name(self, variable):
+        assert (variable is not None and isinstance(variable, ASTVariable)), \
+            '(PyNestML.CodeGeneration.NestReferenceConverter) No or wrong type of uses-gsl provided (%s)!' % type(
+                variable)
+        assert variable.get_scope() is not None, "Undeclared variable: " + variable.get_complete_name()
+
+        variable_name = NestNamesConverter.convert_to_cpp_name(variable.get_complete_name())
+        symbol = variable.get_scope().resolve_to_symbol(variable_name, SymbolKind.VARIABLE)
+        if isinstance(symbol.get_type_symbol(), UnitTypeSymbol):
+            return symbol.get_type_symbol().unit.unit.to_string()
+
+        return ''
+
+    def convert_print_statement(self, function_call):
+        """
+        A wrapper function to convert arguments of a print or println functions
+        :param function_call: print function call
+        :type function_call: ASTFunctionCall
+        :return: the converted print string with corresponding variables, if any
+        :rtype: str
+        """
+        stmt = function_call.get_args()[0].get_string()
+        stmt = stmt[stmt.index('"') + 1: stmt.rindex('"')]  # Remove the double quotes from the string
+        scope = function_call.get_scope()
+        return self.__convert_print_statement_str(stmt, scope)
+
+    def __convert_print_statement_str(self, stmt, scope):
+        """
+        Converts the string argument of the print or println function to NEST processable format
+        Variables are resolved to NEST processable format and printed with physical units as mentioned in model, separated by a space
+
+.. code-block:: nestml
```

Comment:
```suggestion
        .. code-block:: nestml
```

Reply:
To make sure that this renders nicely as HTML, we have to be really pedantic about the whitespace: the code itself is indented three spaces beyond the start of "..", so e.g.:
```
.. code-block:: nestml
   print("Hello world")
```

Converted NEST:
```
.. code-block:: C++
   std::cout << "Hello world";
```